app/sign.js

```diff
@@ -236,9 +243,20 @@ const signEthTx = function(recoveryRequest, key, skipConfirm, isToken) {
     outputs[0].amount = outputs[0].amount.div(TEN.pow(decimals));
   }
 
-  key = promptForConfirmationAndKey(recoveryRequest, outputs, skipConfirm, key);
+  // When generating signatures, we don't currently use EIP155 but this could
+  // be activated if we wanted to. This would protect against replay attacks on other
+  // blockchains, such as Ethereum Classic. To activate the EIP155, we would have to
+  // know the chain ID of the Ethereum blockchains we are using as this value goes
+  // into the V field when using EIP155.
+  // cf. https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+  const useEip155 = false;
```

Comment:
That is really interesting - thx for adding. 

So, when enabling this, would a user have to flip the flag here? If time and/or scope creep weren't an issue, would it make sense to move this type of configuration somewhere else within the app?

Reply:
Do you necessarily need to be able to generate pre-EIP155 signatures? If not, I would just lean towards always using EIP155 signatures. I don't see a reason not to.

That means that this boolean value could be removed and `true` being hardcoded into the `transaction.hash` function call.

This file already contains values like `EOS_MAINNET_CHAIN_ID` and other hardcoded parameters for various blockchains so I imagine that the `CHAIN_ID` values for various Ethereum blockchains could also be contained here. You could then use the `coin` field in your test cases to hold information about which Ethereum blockchain you are signing for.

Reply:
If I knew which testnet blockchain `teth` referred to, I could implement EIP155. But I don't know if you are always using the same test net? I couldn't find any references to `etc` (Ethereum Classic) in this repo, so I don't assume this repo supports Ethereum Classic, so we don't have to worry about that blockchain.

I looked for the txids from the ETH tests on all of etherscan's blockexplorers (main net, Ropsten, Rinkeby, ...) in order to figure out which test net they were published on. But I couldn't find them there so I guess they were never published.

Reply:
We use kovan: https://github.com/BitGo/BitGoJS/blob/master/modules/statics/src/networks.ts#L176